src/Features/Core/Portable/Diagnostics/Analyzers/QualifyMemberAccessDiagnosticAnalyzerBase.cs

```diff
@@ -90,27 +73,18 @@ private void AnalyzeOperation(OperationAnalysisContext context)
             var isQualificationPresent = IsAlreadyQualifiedMemberAccess(memberReference.Instance.Syntax);
             if (shouldOptionBePresent && !isQualificationPresent)
             {
-                DiagnosticDescriptor descriptor;
-                switch (optionValue.Notification.Value)
+                var severity = optionValue.Notification.Value;
+                if (severity != DiagnosticSeverity.Hidden)
                 {
-                    case DiagnosticSeverity.Hidden:
-                        descriptor = null;
-                        break;
-                    case DiagnosticSeverity.Info:
-                        descriptor = s_descriptorQualifyMemberAccessInfo;
-                        break;
-                    case DiagnosticSeverity.Warning:
-                        descriptor = s_descriptorQualifyMemberAccessWarning;
-                        break;
-                    case DiagnosticSeverity.Error:
-                        descriptor = s_descriptorQualifyMemberAccessError;
-                        break;
-                    default:
-                        throw ExceptionUtilities.Unreachable;
-                }
+                    var descriptor = new DiagnosticDescriptor(
```

Comment:
This dynamic creation of DiagnosticDescriptors per reported diagnostic will go away once we have the ability to change our declared SupportedDiagnostics on options changed, so this is only a temporary problem. Plus it's only on open files so it shouldn't be a big deal.

Talked with @mavasani about this offline and we're okay leaving the allocations in for now.